internal/utils/restclientgetter.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import (
+	"fmt"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/discovery"
+	diskcached "k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/util/homedir"
+)
+
+type restClientGetterAdapter struct {
+	config *rest.Config
+}
+
+func RESTClientGetter(kubeConfigPath, kubeContext string) (resource.RESTClientGetter, error) {
+	cfg, err := KubeConfig(kubeConfigPath, kubeContext)
+	if err != nil {
+		return nil, fmt.Errorf("kubernetes client initialization failed: %w", err)
+	}
+
+	return &restClientGetterAdapter{config: cfg}, nil
+}
+
+func (r *restClientGetterAdapter) ToRESTConfig() (*rest.Config, error) {
+	return r.config, nil
+}
+
+func (r *restClientGetterAdapter) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+
+	// The more groups you have, the more discovery requests you need to make.
+	// given 25 groups (our groups + a few custom resources) with one-ish version each, discovery needs to make 50 requests
+	// double it just so we don't end up here again for a while.  This config is only used for discovery.
+	// r.config.Burst = f.discoveryBurst
+
+	cacheDir := filepath.Join(homedir.HomeDir(), ".kube", "cache")
```

Comment:
Flux CLI does not require kubectl to be installed. Will this fail if the `.kube` dir does not exists?

Reply:
I've tested locally, it'll just create the directory. You can also see that [in the code](https://github.com/kubernetes/client-go/blob/v0.22.2/discovery/cached/disk/cached_discovery.go#L175).